src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs

```diff
@@ -131,23 +134,71 @@ private static List<int> GetWhatYouWillNeedSectionsForSequence(int sequenceNumbe
             return sections;
         }
 
-        private static string GetSequenceTitle(int sequenceId)
+
+        [HttpGet("Assessor/Applications/{applicationId}/Sequences/{sequenceNumber}/Sections/{sectionNumber}/Page")]
+        public async Task<AssessorPage> GetFirstAssessorPage(Guid applicationId, int sequenceNumber, int sectionNumber)
         {
-            switch (sequenceId)
+            return await GetAssessorPage(applicationId, sequenceNumber, sectionNumber, null);
+        }
+
+        [HttpGet("Assessor/Applications/{applicationId}/Sequences/{sequenceNumber}/Sections/{sectionNumber}/Page/{pageId}")]
+        public async Task<AssessorPage> GetAssessorPage(Guid applicationId, int sequenceNumber, int sectionNumber, string pageId)
+        {
+            AssessorPage page = null;
+
+            var qnaSection = await _qnaApiClient.GetSectionBySectionNo(applicationId, sequenceNumber, sectionNumber);
+            var qnaPage = qnaSection?.QnAData.Pages.FirstOrDefault(p => p.PageId == pageId || string.IsNullOrEmpty(pageId));
+
+            if (qnaPage != null)
```

Comment:
Is it worth moving this code block into a service or orchestrator?

Reply:
This block of code has been refactored